src/app/core/utils/math/ClampedValue.ts

```diff
@@ -47,15 +47,20 @@ export class ClampedValue {
      */
     public constructor(initialValue?: number, minValue?: number, maxValue?: number) {
         this.value = initialValue;
-        this.minValue = minValue || initialValue; // if min not given use initial
-        this.maxValue = maxValue || initialValue; // if max not given use initial
+        this.minValue = minValue ?? initialValue; // if min not given use initial
+        this.maxValue = maxValue ?? initialValue; // if max not given use initial
     }
 
     /**
      * Update the number to a new value, or the closest clamp bound if outside allowed range.
      * @param val The clamped value of the number
+     * @throws {Error} If this.minValue or this.maxValue are undefined
      */
     public setValue(val: number): void {
+        if (this.minValue === undefined)
+            throw new Error("ClampedValue.setValue failed: this.minValue is undefined");
+        if (this.maxValue === undefined)
+            throw new Error("ClampedValue.setValue failed: this.minValue is undefined");
```

Comment:
Blech yeah okay I think this was a mistake, I apologize for even recommending it - definitely just unnecessary bloat and really points towards the poor design of the model